syntax = "proto3";

package sekai.db;

import "sekai/db/proto/descriptor.proto";
import "sekai/db/proto/enums.proto";

message AreaItemLevel {
  option (master_db_file)                    = "areaItemLevels.json";
  optional int32 area_item_id                = 1 [(json_name) = "areaItemId", (primary_key) = true];
  optional int32 level                       = 2 [(json_name) = "level"];
  optional Unit target_unit                  = 3 [(json_name) = "targetUnit", (string_empty_value) = "any"];
  optional Attr target_card_attr             = 4 [(json_name) = "targetCardAttr", (string_empty_value) = "any"];
  optional int32 target_game_character_id    = 5 [(json_name) = "targetGameCharacterId"];
  optional float power1_bonus_rate           = 6 [(json_name) = "power1BonusRate"];
  optional float power1_all_match_bonus_rate = 7 [(json_name) = "power1AllMatchBonusRate"];
  optional float power2_bonus_rate           = 8 [(json_name) = "power2BonusRate"];
  optional float power2_all_match_bonus_rate = 9 [(json_name) = "power2AllMatchBonusRate"];
  optional float power3_bonus_rate           = 10 [(json_name) = "power3BonusRate"];
  optional float power3_all_match_bonus_rate = 11 [(json_name) = "power3AllMatchBonusRate"];
  optional string sentence                   = 12 [(json_name) = "sentence"];
}
